src/main/java/net/sf/jabref/gui/importer/fetcher/ACMPortalFetcherGUI.java

```diff
@@ -0,0 +1,232 @@
+
+        ButtonGroup group = new ButtonGroup();
+        group.add(acmButton);
+        group.add(guideButton);
+
+        pan.add(absCheckBox);
+        pan.add(acmButton);
+        pan.add(guideButton);
+
+        return pan;
+    }
+
+    @Override
+    public boolean processQueryGetPreview(String query, FetcherPreviewDialog preview, OutputPrinter status) {
+        setTerms(query);
+        setPiv(0);
+        setShouldContinue(true);
+        setAcmOrGuide(acmButton.isSelected());
+        setFetchAbstract(absCheckBox.isSelected());
+        String address = makeUrl();
+        LinkedHashMap<String, JLabel> previews = new LinkedHashMap<>();
+
+        try {
+            URL url = new URL(address);
+
+            URLDownload dl = new URLDownload(url);
+
+            String page = dl.downloadToString();
+
+            int hits = getNumberOfHits(page, getResultsFoundPattern(), ACMPortalFetcher.getHitsPattern());
+
+            int index = page.indexOf(getResultsFoundPattern());
+            if (index >= 0) {
+                page = page.substring(index + getResultsFoundPattern().length());
+            }
+
+            if (hits == 0) {
+                status.showMessage(Localization.lang("No entries found for the search string '%0'", getTerms()),
+                        Localization.lang("Search %0", getTitle()), JOptionPane.INFORMATION_MESSAGE);
+                return false;
+            } else if (hits > 20) {
+                status.showMessage(
+                        Localization.lang("%0 entries found. To reduce server load, only %1 will be downloaded.",
+                                String.valueOf(hits), String.valueOf(getPerPage())),
+                        Localization.lang("Search %0", getTitle()), JOptionPane.INFORMATION_MESSAGE);
+            }
+
+            hits = getNumberOfHits(page, getPageRangePattern(), getMaxHitsPattern());
+            parse(page, Math.min(hits, getPerPage()), previews);
+            for (Map.Entry<String, JLabel> entry : previews.entrySet()) {
+                preview.addEntry(entry.getKey(), entry.getValue());
+            }
+            return true;
+
+        } catch (MalformedURLException e) {
+            getLogger().warn("Problem with ACM fetcher URL", e);
+        } catch (ConnectException e) {
+            status.showMessage(Localization.lang("Could not connect to %0", getTitle()),
+                    Localization.lang("Search %0", getTitle()), JOptionPane.ERROR_MESSAGE);
+            getLogger().warn("Problem with ACM connection", e);
+        } catch (IOException e) {
+            status.showMessage(e.getMessage(), Localization.lang("Search %0", getTitle()), JOptionPane.ERROR_MESSAGE);
+            getLogger().warn("Problem with ACM Portal", e);
+        }
+        return false;
+
+    }
+
+    @Override
+    public HelpFiles getHelpPage() {
+        return HelpFiles.FETCHER_ACM;
+    }
+
+    /**
+     * Find out how many hits were found.
+     * @param page
```

Comment:
what is with marker and pattern? maybe you could also describe very short what marker and pattern are